infrastructure/lib/kit-infrastructure.ts

```diff
@@ -45,6 +46,7 @@ export class KITInfrastructure extends Stack {
       clusterName: id,
       vpc: vpc,
       role: clusterRole,
+      kubectlLayer: new KubectlLayer(this, 'kubectl'),
```

Comment:
Why do we need this?

Reply:
we need this to make sure `kubectl` is selected as the same version of the cluster, otherwise it will be defaulted to `1.20` and that will not respect version skew and cause issues.